### PR TITLE
SAK-49656 T&Q: Ability to associate an exam with an existing GB item

### PR DESCRIPTION
**Pull request by mpellicer**

### CODE REVIEW

File: samigo/samigo-hibernate/src/java/org/sakaiproject/tool/assessment/data/dao/assessment/AssessmentBaseData.java [hunk 1]
```diff
@@ -356,7 +357,7 @@ public String getAssessmentMetaDataByLabel(String label) {
   }
 
   public void addAssessmentMetaData(String label, String entry) {
-    if (this.assessmentMetaDataMap.get(label)!=null){
+      if (this.assessmentMetaDataMap.containsKey(label)) {
```
Comment (bgarciaentornos) [inline samigo/samigo-hibernate/src/java/org/sakaiproject/tool/assessment/data/dao/assessment/AssessmentBaseData.java:360]: Bad spacing

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/GradebookServiceHelperImpl.java [hunk 2]
```diff
@@ -196,28 +211,59 @@ public void updateExternalAssessmentScore(AssessmentGradingData ag,
...
-      ag.getAgentId(),  points);
     if (testErrorHandling){
       throw new Exception("Encountered an error in update ExternalAssessmentScore.");
     }
   }
-  
+
   public void updateExternalAssessmentScores(Long publishedAssessmentId, final Map<String, Double> studentUidsToScores,
-		  org.sakaiproject.grading.api.GradingService g) throws Exception {
+    org.sakaiproject.grading.api.GradingService g) throws Exception {
```
Comment (bgarciaentornos) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/GradebookServiceHelperImpl.java:245]: Spacing

**Review by bgarciaentornos — CHANGES_REQUESTED** [review]
Some minor changes.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SavePublishedSettingsListener.java [hunk 3]
```diff
@@ -854,114 +880,126 @@ public boolean checkScore(PublishedAssessmentSettingsBean assessmentSettings, Pu
...
+        // a. if Gradebook does not exists, do nothing, 'cos setting should have been hidden
+        // b. if Gradebook exists, just call addExternal and removeExternal and swallow any exception. The
+        //    exception are indication that the assessment is already in the Gradebook or there is nothing
+        //    to remove.
+        org.sakaiproject.grading.api.GradingService gradingService = null;
+        if (integrated) {
+            gradingService = (org.sakaiproject.grading.api.GradingService) SpringBeanLocator.getInstance().getBean("org.sakaiproject.grading.api.GradingService");
+        }
+
+        PublishedEvaluationModel evaluation = (PublishedEvaluationModel) assessment.getEvaluationModel();
```
[the remark this answers is not in the document]
Comment (mpellicer) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SavePublishedSettingsListener.java:894]: Please don't mention it, I think I fixed it but also changes the spacing in the function name.

image

File: samigo/samigo-hibernate/src/java/org/sakaiproject/tool/assessment/data/dao/assessment/AssessmentBaseData.java (hunk 1 above)
Comment (mpellicer) [reply samigo/samigo-hibernate/src/java/org/sakaiproject/tool/assessment/data/dao/assessment/AssessmentBaseData.java:360]: Nooooooooooooo

image

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/GradebookServiceHelperImpl.java (hunk 2 above)
Comment (mpellicer) [reply samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/GradebookServiceHelperImpl.java:245]: No man, it's fine

Comment (bgarciaentornos) [reply samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/GradebookServiceHelperImpl.java:245]: It has the same indentation that the next line, it looks wrong tbh

File: samigo/samigo-hibernate/src/java/org/sakaiproject/tool/assessment/data/dao/assessment/AssessmentBaseData.java (hunk 1 above)
Comment (bgarciaentornos) [reply samigo/samigo-hibernate/src/java/org/sakaiproject/tool/assessment/data/dao/assessment/AssessmentBaseData.java:360]: Not sure I understand the screenshot, surely that "if" and the following lines shouldn't have the same indentation?

Comment (mpellicer) [reply samigo/samigo-hibernate/src/java/org/sakaiproject/tool/assessment/data/dao/assessment/AssessmentBaseData.java]: You're right, reverted

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/GradebookServiceHelperImpl.java (hunk 2 above)
Comment (mpellicer) [reply samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/GradebookServiceHelperImpl.java]: Well, I reverted this and now looks greater in Github, you okay then?

**Review by bgarciaentornos — APPROVED** [review]

**Pull request merged**